### PR TITLE
Improved `irictl machine` get events command

### PR DESCRIPTION
**Pull request by ushabelgur**
# Proposed Changes

- Added time filter to get events in irictl command

### CODE REVIEW

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go [hunk 1]
```diff
@@ -17,11 +18,15 @@ import (
...
-	Labels map[string]string
+	Labels         map[string]string
+	EventsFromTime string
+	EventsToTime   string
 }
 
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.StringToStringVarP(&o.Labels, "labels", "l", o.Labels, "Labels to filter the events by.")
+	fs.StringVarP(&o.EventsFromTime, "eventsFromTime", "f", o.EventsFromTime, fmt.Sprintf("Events From Time to filter the events by. In the format of %s", time.DateTime))
+	fs.StringVarP(&o.EventsToTime, "eventsToTime", "t", o.EventsToTime, fmt.Sprintf("Events To Time to filter the events by. In the format of %s", time.DateTime))
```
Comment (lukasfrank) [inline irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go:29]: I would propose to drop the `ToTime` and always set it to the current time.

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go [hunk 2]
```diff
@@ -17,11 +18,15 @@ import (
...
 type Options struct {
-	Labels map[string]string
+	Labels         map[string]string
+	EventsFromTime string
+	EventsToTime   string
 }
 
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.StringToStringVarP(&o.Labels, "labels", "l", o.Labels, "Labels to filter the events by.")
+	fs.StringVarP(&o.EventsFromTime, "eventsFromTime", "f", o.EventsFromTime, fmt.Sprintf("Events From Time to filter the events by. In the format of %s", time.DateTime))
```
Comment (lukasfrank) [inline irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go:28]: Can we use a `Duration` here and default it to some value?

**Review by lukasfrank — CHANGES_REQUESTED** [review]

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go (hunk 2 above)
Comment (ushabelgur) [reply irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go]: Done added duration with default value of 60 minutes

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go (hunk 1 above)
Comment (ushabelgur) [reply irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go]: Done

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go [hunk 3]
```diff
@@ -16,12 +17,16 @@ import (
...
+
 type Options struct {
-	Labels map[string]string
+	Labels   map[string]string
+	Duration string
 }
 
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.StringToStringVarP(&o.Labels, "labels", "l", o.Labels, "Labels to filter the events by.")
+	fs.StringVarP(&o.Duration, "duration", "d", o.Duration, "Duration to filter the events by.")
```
Comment (lukasfrank) [inline irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go:29]: Please use `DurationVarP`

**Review by lukasfrank — CHANGES_REQUESTED** [review]

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go (hunk 3 above)
Comment (ushabelgur) [reply irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go]: Done

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go [hunk 4]
```diff
@@ -68,11 +73,16 @@ func Run(
...
 	if opts.Labels != nil {
-		filter = &iri.EventFilter{
-			LabelSelector: opts.Labels,
-		}
+		filter.LabelSelector = opts.Labels
+	}
+
+	if opts.Duration > 0 {
+		filter.EventsFromTime = time.Now().Add(-1 * opts.Duration).Unix()
 	}
```
Comment (lukasfrank) [inline irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go:86]: This is not needed if Options.Duration always contains the actual duration

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go [hunk 5]
```diff
@@ -68,11 +73,16 @@ func Run(
...
+	var filter *iri.EventFilter = &iri.EventFilter{
+		EventsFromTime: time.Now().Add(-1 * DefaultDuration).Unix(),
+		EventsToTime:   time.Now().Unix(),
+	}
 	if opts.Labels != nil {
-		filter = &iri.EventFilter{
-			LabelSelector: opts.Labels,
-		}
+		filter.LabelSelector = opts.Labels
+	}
```
Comment (lukasfrank) [inline irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go:82]: Do we need this check? I think we can always set opts.Labels, right?

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go [hunk 6]
```diff
@@ -16,12 +17,16 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+const DefaultDuration = 60 * time.Minute
```
[the remark this answers is not in the document]
Comment (ushabelgur) [reply irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go]: done

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go (hunk 4 above)
Comment (ushabelgur) [reply irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go]: removed check

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go (hunk 5 above)
Comment (ushabelgur) [reply irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go]: not needed removed

File: irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go [hunk 7]
```diff
@@ -16,12 +17,16 @@ import (
...
+
 type Options struct {
-	Labels map[string]string
+	Labels   map[string]string
+	Duration time.Duration
```
[the remark this answers is not in the document]
Comment (ushabelgur) [reply irictl-machine/cmd/irictl-machine/irictlmachine/get/event/event.go:22]: added default value in fs.DurationVarP itself

**Review by lukasfrank — APPROVED** [review]

**Pull request merged**